frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/SelectedNodeInputOutputTab.tsx

```diff
@@ -19,6 +22,16 @@ const SelectedNodeInputOutputTab: React.FC<SelectedNodeInputOutputTabProps> = ({
   task,
   execution,
 }) => {
+  const { namespace } = usePipelinesAPI();
+
+  const getExecutionDisplayNameFromObject = React.useCallback(
+    () =>
+      execution?.customPropertiesMap.find(
+        ([customPropKey]) => customPropKey === 'display_name',
+      )?.[1].stringValue || '(No name)',
```

Comment:
this already exists in a util file `getExecutionDisplayName` at `src/pages/pipelines/global/experiments/executions/utils.ts`

Reply:
Yes, but that takes execution as `Execution` type, on this page we have `Execution.AsObject` as the type.

Reply:
i see. i would just switch the type in the props to just be Execution and then adjust the get execution fields map to first convert it to an object before doing anything with it:

`execution?.toObject()?.customPropertiesMap.find(([customPropKey]) => customPropKey === key)?.[1]`

Reply:
but then in the unit test, how can we pass data as Execution?